AutonomousDemo/VehicleMotion.cs

```diff
@@ -25,7 +27,7 @@ public Velocity CalculateCurrentVelocity(double meters, int seconds)
             return new Velocity(meters/seconds);
         }
 
-        public Velocity DecelerateVelocityPerSecond(Velocity velocity)
+        public Velocity Decelerate(Velocity velocity)
```

Comment:
This (and Accelerate(...) below) both has a side effect on the passed-in Velocity _and_ returns it. The returned value is never used. Probably a safer thing to do would be to return a _new_ Velocity and make use of it.

In the long run it might also be beneficial to make Velocity a struct, i.e. a value type, so you can't actually modify an instance of it.